packages/sanity/src/core/config/studio/types.ts

```diff
@@ -23,6 +23,10 @@ export interface LogoProps {
  * @beta */
 export interface NavbarProps {
   renderDefault: (props: NavbarProps) => ReactElement
+  /**
+   * @internal
+   * @beta */
+  __internal_rightSectionNode?: ReactNode
```

Comment:
We should probably think about how we could offer an API with more granularity, so we could use a render/renderDefault pattern for these things as well, but I understand why `navbar.render` + `renderDefault` didn't work for you here.

Reply:
Yes, it was an interesting discussion with the ex team, but we couldn't find any other solution given this navbar component adds multiple internal components and we would have need to recreate the same through the plugin. 
Will be a nice discussion to bring to next studio meeting. Will propose it. 

Thanks @bjoerge !